src/agent/proxy.c

```diff
@@ -0,0 +1,327 @@
+                        active_state_str,
+                        substate,
+                        reason);
+
+        if (proxy->request_message) {
+                /* We're waiting for the initial start-or-fail to report back
+                 *
+                 * There are two main things that can happen here.
+                 *
+                 * Either the state eventually changes to any kind of
+                 * active state. Then we consider the target service
+                 * to have started, and we report that, and switch to waiting
+                 * for it to stop. It is possible that we get a virtual
+                 * active state change because the service was already running,
+                 * but this is fine here.
+                 *
+                 * Alternatively, we get a message that says the state
+                 * switched to failed, or inactive. This means the service
+                 * stopped, which we report as a failure to start the target.
+                 * However, here we have to be careful about virtual inactive
+                 * events, as we may get a virtual inactive for the current
+                 * state before then getting a real transition to the new state.
+                 *
+                 * In addition, it is possible that we never ever change state, because
+                 * the service fails to parse, for instance. If so, we hit the (real) removed
+                 * event before seeing any activation. See below for that.
+                 */
+
+                if (active_state == UNIT_ACTIVE) {
+                        proxy_service_initial_state_reached(proxy, true);
+                } else if ((active_state == UNIT_FAILED || active_state == UNIT_INACTIVE) &&
+                           streq(reason, "real")) {
+                        proxy_service_initial_state_reached(proxy, false);
+                }
+        } else if (proxy->sent_successful_ready) {
+                /* We reached the initial state and are now monitoring for the target to exit */
+                if ((active_state == UNIT_FAILED || active_state == UNIT_INACTIVE) && streq(reason, "real")) {
+                        proxy_service_target_stopped(proxy);
+                }
+        }
+
+        return sd_bus_reply_method_return(m, "");
+}
+
+static int proxy_service_method_new(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *ret_error) {
+        _cleanup_proxy_service_ ProxyService *proxy = proxy_service_ref((ProxyService *) userdata);
+        Agent *agent = proxy->agent;
+        if (agent == NULL) {
+                return sd_bus_reply_method_errorf(m, SD_BUS_ERROR_FAILED, "Internal error");
+        }
+
+        const char *reason = NULL;
+        int r = sd_bus_message_read(m, "s", &reason);
+        if (r < 0) {
+                return sd_bus_reply_method_errorf(m, SD_BUS_ERROR_INVALID_ARGS, "Invalid arguments");
+        }
+
+        hirte_log_errorf("Got proxy new: %s", reason);
+
```

Comment:
Do we need to `proxy_service_initial_state_reached(proxy, false);` here as well?

Reply:
No, we always get a new event when something is starting to activate. Basically when systemd instantiates the unit in memory. We don't yet know anything about how it activates.